### ISSUE TITLE
We have a `global` scope. Why don't we have a `module` scope?

### ISSUE THREAD

**Issue by papiando** [nodejs/help#3045]
I was trying to publish some objects I have in a namespace in the module scope, but have been unsuccessful. It is actually possible to publish the objects using `global`, but there is no way to use, for example `module`. I'll give an example.

The namespace shown loads objects dynamically based on whatever features are available. The result is a namespace like this:

```javascript
[class Namespace] {
  Class1: [class Class1],
  SubclassA: [class SubclassA extends Class1],
  SubclassB: [class SubclassB extends Class1]
}
```

I know I can access the classes through `new Nameclass.SubclassA()` for example. I can also publish these to `global`:

```javascript
import Namespace from './Namespace.mjs';
for(const obj of Object.keys(Namespace)) {
  global[obj] = Namespace[obj];
}
```

However, this would make the objects available throughout all the modules. If I want to confine this to the module scope, there seems to be no way to do this. Ultimately, it would be great to do something like this:

```javascript
import Namespace from './Namespace.mjs';
for(const obj of Object.keys(Namespace)) {
  moduleScope[obj] = Namespace[obj];
}
```

My guess is that it wouldn't be so hard to do. Perhaps even using `module`? Although that might conflict with the CommonJS variant. Anyway, it seems consistent with the `global` to have something like `module` to access the scope on a module level.

**Comment by aduh95** [comment]
To be clear, what you would like to do is:

```js
import Namespace from "./Namespace.mjs";

/* some magic code here */

new SubClassA(); // SubClassA here refers to Namespace.SubClassA
globalThis.SubClassA === undefined // yet SubClassA is not on the global scope
```

I don't think this is achievable with the current ECMAScript spec, [`with` statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/with) would actually do the trick for what you're trying to achieve, but they are forbidden in ES6 modules – for good reasons. Node.js is very unlikely not going to introduce a language feature, that would need to go through a [TC39 proposal](https://github.com/tc39/proposals) first, and being implemented by browsers as well.

**Comment by papiando** [comment]
Yes, correct. `with` statements would have worked excellent. So would some kind of `use(Namespace)` possibility. I'll check up on the [TC39 proposal](https://github.com/tc39/proposals).
